Indexes/CurrencyPartIndex.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using OrchardCore.Commerce.Models;
+using OrchardCore.Commerce.Money;
+using OrchardCore.ContentManagement;
+using YesSql.Indexes;
+
+namespace OrchardCore.Commerce.Indexes
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class CurrencyPartIndex : MapIndex
```

Comment:
Why do we need an index on currencies? That seems wrong to me, especially as there may be multiple `ICurrencyProvider` implementations all contributing currencies.

Reply:
I added the index so that it would be quick to find the currency documents when it's stored as content items.

I ran into some issues when testing with loading multiple CurrencyProviders at the same time. I'll write more about that below in the main thread.